### PR TITLE
When constructing a monitored resource, translate namespace_name and pod_name to namespace_id and pod_id, respectively.

### PR DESCRIPTION
**Pull request by igorpeshansky**
@crassirostris @summit FYI.

### CODE REVIEW

**Comment by crassirostris** [comment]
Thanks, LGTM

File: test/plugin/base_test.rb [hunk 1]
```diff
@@ -270,23 +270,23 @@ def setup
...
         'instance_id' => VM_ID,
+        'pod_id' => CONTAINER_POD_NAME,
         'container_name' => CONTAINER_CONTAINER_NAME,
         'zone' => ZONE
       }
     },
     log_name: CONTAINER_CONTAINER_NAME,
     project_id: PROJECT_ID,
     labels: {
-      "#{CONTAINER_CONSTANTS[:service]}/namespace_name" =>
```
Comment (qingling128) [inline test/plugin/base_test.rb]: Is removing these two labels going to break anything?

Comment (igorpeshansky) [inline test/plugin/base_test.rb]: That's exactly the question that @summit was planning to answer.

File: lib/fluent/plugin/out_google_cloud.rb [hunk 2]
```diff
@@ -450,12 +450,9 @@ def compute_group_resource_and_labels(tag)
...
         if match_data
-          group_resource.labels['container_name'] =
-            match_data['container_name']
-          %w(namespace_name pod_name).each do |field|
-            group_common_labels["#{CONTAINER_CONSTANTS[:service]}/#{field}"] =
-              match_data[field]
-          end
+          group_resource.labels['container_name'] = match_data['container_name']
+          group_resource.labels['namespace_id'] = match_data['namespace_name']
+          group_resource.labels['pod_id'] = match_data['pod_name']
```
Comment (qingling128) [inline lib/fluent/plugin/out_google_cloud.rb]: BTW, if `pod_id` and `namespace_id` are present in the `kubernetes` field in the log entry, the values of resource labels `pod_id` and `namespace_id` here would be overwritten. Do we want that? Sounds like the `kubernetes` field is to be deprecated or something (I just read the original ticket).

Comment (crassirostris) [inline lib/fluent/plugin/out_google_cloud.rb]: > Sounds like the `kubernetes` field is to be deprecated or something

It's not used in the fluentd for GCP deployment, yes

Comment (qingling128) [inline lib/fluent/plugin/out_google_cloud.rb]: @igorpeshansky: Should we clean it up then? A follow up ticket or TODO is fine.

Comment (igorpeshansky) [inline lib/fluent/plugin/out_google_cloud.rb]: @qingling128: That would involve deprecating (support for) the [kubernetes_metadata_filter](https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter) plugin. I would rather tackle that separately. As things stand, the `container_name` field from the `kubernetes` record will have been overwritten by this code already -- we're just adding a couple more.

Comment (qingling128) [inline lib/fluent/plugin/out_google_cloud.rb]: Ah, good to know about the filter plugin.

For the overwriting part, I think it's the other way around though. Data extracted from `tag` is set when processing group label level, while data extracted from `kubernetes` record is set later on when processing entry level labels. But I guess the order doesn't really matter If a user does decide to specify different values in different places...

**Comment by qingling128** [comment]
LGTM

**Comment by igorpeshansky** [comment]
Waiting for @summit to confirm that deleting the labels is harmless.

**Comment by igorpeshansky** [comment]
Ping @summit.

**Comment by igorpeshansky** [comment]
As discussed with @summit, we need to leave the labels around as users are actually filtering on them. We'll deprecate and remove them later.

File: test/plugin/base_test.rb (hunk 1 above)
Comment (igorpeshansky) [inline test/plugin/base_test.rb]: You were right -- @summit found some uses of these, so we're leaving them in. Pushed a new commit. PTAL.

**Comment by igorpeshansky** [comment]
Thanks all. Merging.

**Pull request merged**